### ISSUE TITLE
Add submission type section: Mobile app, AI project

### ISSUE THREAD

**Issue by rsgonzal** [Iridescent-CM/technovation-app#3011]
As a student, I need to indicate whether I submitted a mobile app or AI project
- Add under "Code" section
- Title: "Submission Type"
- Help text: "_Please indicate which type of submission this is. This will affect how you are judged. Think back on the work you did for these lessons: Beginner Division OR Junior/Senior Division._"
- Choices: "Mobile App" & "AI Project"
- Add button to save/ Edit (based off conventions, i think "edit" is used on overview of section and then "save" could be used when you click in to that section
- Add tip to bottom of box (smaller & in italics): "_Tip: If most of the work you did was training an AI model, choose "AI Submission". If most of the work you did was coding an app, choose "Mobile App Submission"._"
- Should also be included in summary page

<img width="582" alt="Screen Shot 2021-12-21 at 10 17 57 PM" src="https://user-images.githubusercontent.com/22801026/147045064-de07f5e7-7b0e-4c14-8641-53a67beab5fc.png">

**Comment by nickiredescent** [comment]
I'm not in love with "Upload your Source Code" since it's not really accurate for AI submissions. I've called this section "Technical Additions" in the curriculum/other documents. Would like someone from PT to reconfirm how we want to name this.


Here's some text we can include to clarify what should be included in AI submissions:

> **AI Project Submissions**
> AI submissions should be a zip file with the following included:
> - The dataset you used to train your model
> - This can be a screenshot, spreadsheet, or direct link to an image/sound folder, etc.
> - Picture(s) of prototypes
> - This can be cardboard models, drawings, sample devices, etc.
> - (For online inventions) A link with sample login information

I understand that we can't have individual fields with proper checks for all these things. We're just asking for a zip file this year and hopefully the text is enough.

**Comment by ksrebecca** [comment]
Why don't we change "Upload your source code" to "Upload your technical work"

agree with nick's text edits

when they are forced to choose mobile or AI on that screen can we also add a sentence to the end of the help text that says, "Click here if you aren't sure which to choose." - @nickiredescent can you work with Karen to quickly (early next week) develop a checklist or something easily comprehensible super fast that summarizes why they would choose one over the other? Karen's slide from the ChA AI training this week could be a good starting point

**Comment by ophelieh** [comment]
I do like the name: "Technical Additions" or "Technical Elements" for the name of this section. And love the new design!!!

**Comment by nickiredescent** [comment]
Add to help text:

> Think back on the work you did for these lessons: [Beginner Division](https://beginner.technovationchallenge.org/path-player?courseid=beginners&unit=beginners_1622560910003_0Unit) OR [Junior/Senior Division](https://technovationchallenge.org/curriculum/ideation-10/).

> _Tip: If most of the work you did was training an AI model, choose "AI Submission". If most of the work you did was coding an app, choose "Mobile App Submission"._

I wouldn't want to add more than this and get overly wordy for our international audience.

**Comment by rsgonzal** [comment]
Here is an update mockup from what I understood
<img width="582" alt="Screen Shot 2021-12-21 at 10 17 57 PM" src="https://user-images.githubusercontent.com/22801026/147045064-de07f5e7-7b0e-4c14-8641-53a67beab5fc.png">

**Comment by rsgonzal** [comment]
opened #3208 for source code page edits. Needs more refinement

**Comment by ophelieh** [comment]
I like the help text! I do think we need to change the title for this section as it is no longer relevant. 

We could go with: "Technical Additions" (like when they are uploading their work), "Technical Elements" (which might be easier for a quick glance and as a title)

**Comment by ksrebecca** [comment]
also like the help text. yes @rsgonzal your understanding is great plus echoing @ophelieh to change both the section name "Code" > "Technical Elements" and the bullet point "Upload your source code" > "Technical Additions"

**Comment by rsgonzal** [comment]
selecting AI project doesn't seem to be saving

**Comment by shaunxp20** [comment]
> selecting AI project doesn't seem to be saving

It should be working now, I added the submission type (AI Project or Mobile App) to the "Code" summary section, so it should be easier to tell if it's saving or not. 🤞😄

**Comment by ophelieh** [comment]
* Edit title of section to read "Technical Elements" instead of Code. And edit "Submission: Code" to read "Submission: Technical Elements" 
![Screen Shot 2022-01-13 at 12 16 29 PM (2)](https://user-images.githubusercontent.com/32206421/149386586-a9ce892b-4970-4b8d-943c-97967fd14691.png)


* Replace "1. Development Platform" with "1. Submission Type" 

* AI still doesn't seem to save. It tells me there is nothing there. But when I click on "Select your submission type", it shows I selected AI. 
![Screen Shot 2022-01-13 at 12 12 38 PM (2)](https://user-images.githubusercontent.com/32206421/149385874-210ea5df-875b-4692-82a4-934517dfc167.png)

**Comment by ophelieh** [comment]
I went back to edit from selecting Mobile App and uploading my materials and changed to AI. This time it seemed to save it - does that mean it was fixed or is that a bug? 

Note to also change the name from Code to "Technical Elements" in the button below Images

![Screen Shot 2022-01-13 at 12 35 50 PM (2)](https://user-images.githubusercontent.com/32206421/149389148-4bb128af-af71-46a7-88ae-a8567ed456e9.png)

**Comment by rsgonzal** [comment]
https://user-images.githubusercontent.com/22801026/149462867-398d6ef8-1b57-4f71-a4ff-4224f69ec9ae.mov

**Comment by rsgonzal** [comment]
Very weird. Like Ophelie mentioned, if you 
- first select mobile app and save
- go back and change it to AI project
- hit save
- now it let's me upload source code and keeps the check mark on the menu bar

**Comment by dboyer** [comment]
I was logged in as a mentor this morning and had the same issue Roxana showed in the video. Not sure if this is helpful, but I found:

1. Select AI and get the loop issue with not being able to add the code
2. Change my submission type to mobile app and can fill out all the fields as requested including a link to a Thunkable project
3. Thunkable link automatically populates in the technical additions project URL field
4. Change my submission type to AI and it saves
5. Thunkable project link remains in technical additions fields
<img width="735" alt="Screen Shot 2022-01-14 at 10 47 11 AM" src="https://user-images.githubusercontent.com/516764/149543945-d7d8e686-5107-4c3b-8f7e-08589576ff47.png">
5. Change the technical additions to a general URL
<img width="711" alt="Screen Shot 2022-01-14 at 10 48 00 AM" src="https://user-images.githubusercontent.com/516764/149544077-746b2617-1a86-46c2-9e3c-ae759cc198ff.png">
6. New URL does not update on the main screen but when I click back to the upload screen (#5 above) the new URL is there
<img width="702" alt="Screen Shot 2022-01-14 at 10 48 19 AM" src="https://user-images.githubusercontent.com/516764/149544141-6c354979-dc38-4210-9918-d17dfa59c1f4.png">

Long story short, there's some type of interaction in the technical additions URL and the URL for mobile apps that is interacting strangely.

**Comment by rsgonzal** [comment]
Hmm interesting, deb! it seems to still think it's thunkable. Because no other option should require a URL 🤔

**Comment by shaunxp20** [comment]
Okay, I made some changes that I think will fix most of these issues, there's a lot at play so I'm only like 85% confident I addressed all of them! Anywho, the changes are on QA. 😄

**Comment by rsgonzal** [comment]
so it looks like it saved it, but i didn't get a checkmark 😅

**Comment by rsgonzal** [comment]
no checkmark and won't let me submit!

**Comment by shaunxp20** [comment]
> so it looks like it saved it, but i didn't get a checkmark 😅
> no checkmark and won't let me submit!

Great catch on both of these! I finally got a couple of fixes on QA, hopefully that fixes them! 🤞

**Comment by rsgonzal** [comment]
Seems to be working for me! submitted some of my previous apps and started a new one. AI works and switching to mobile app works too

**Comment by dboyer** [comment]
This is working for me as well. I think there's still something going on with the platform remembering the previous URL/file upload when you switch between submission types, but it doesn't seem to impact the final data that is uploaded which is most important.
